### DISCUSSION TITLE
Code quality

### DISCUSSION THREAD

**Discussion by filipkowal** [TheIOFoundation/ProjectLockdown#204] [Ideas] [answered]
Here is my list of proposed rules to which we could adhere when writing and refactoring code. I tried to make it short so I can check it quickly before every push. Please, share your opinions and propose new rules which you feel are important and relevant.

* [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) (Don't repeat yourself)
* [KISS](https://en.wikipedia.org/wiki/KISS_principle) (Keep it simple, stupid)
* Scout rule. Leave the campground cleaner than you found it.
* Always fix the root cause of the problem instead of symptoms treatment.
* Be consistent. If you do something a certain way, do all similar things in the same way.
* Use explanatory, unambigous names
* Replace magic numbers with named constants.
* Make functions small, do one thing
* Use components' composition to make components smaller [react example](https://reactjs.org/docs/thinking-in-react.html)

Most of them where taken from this summary of "Clean Code" - the book: https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29

**Comment by jeff-knurek** [comment, marked as answer]
I think these are great pieces of advice @filipkowal 
and like Jean suggested, I think these can go right to the wiki in: https://github.com/TheIOFoundation/ProjectLockdown/wiki/Dev-Guidelines#Project-Lockdown-Specific-Dev-Guidelines

**Reply by filipkowal** [reply]
And so they did and stayed there happily forever after.
